### DISCUSSION TITLE
Support versionned presets

### DISCUSSION THREAD

**Discussion by michael-todorovic** [renovatebot/renovate#11258] [Suggest an Idea]
Hello,
I read the dev doc so here I am ;)
We're happy users of renovatebot and recently switched to shared presets. This is working great but there's one thing I find missing. From what I understood in the code, we can't really specify a tag/version. It's only getting files from the repository default branch. It would be supernice for reliability to be able to use tags as well.

Do you find it would be a nice addition? I'm not confortable with typescript yet but I could try to create a PR for that feature :)

Thanks!

**Comment by rarkins** [comment]
First step would be to work through the requirements. I'd include:
- We need to find a sensible way of expressing the tag/version within our existing syntax (I think we unfortunately use `#` already so it needs disambiguation)
- We need to double check that we can fetch such tagged versions efficiently (i.e. without too many API requests or cloning the entire repo)
- It's not essential for the initial version, but I guess we need a Renovate presets manager added, so it can self-update

Feel free to open a feature request issue, especially if you're interested in trialing it yourself.

**Reply by michael-todorovic** [reply]
I don't see the `#` from https://github.com/renovatebot/renovate/blob/main/docs/development/shareable-configs.md nor https://docs.renovatebot.com/config-presets/#gitlab-hosted-presets, it seems a good candidate (but I may have not seen something in the code, I'm not used to it yet). Otherwise, I thought about `=`

I would try to add the manager in the PR, it will depend on the time I can dedicate on this. If I can't now, I could simply push a doc update with regexManager till we have it

**Reply by rarkins** [reply]
I need to check that there's not any other use cases we missed there. I thought we had the concept of subdirectories too.

**Reply by michael-todorovic** [reply]
I did some testing and this works well with github, thanks to `?ref=myTag`. Gitlab has also this `ref` support so that should be ok as well. I haven't so much experience with npm renovate packages (not using them at all) so this is my next test.

My first tests were with 
```
    "github>michael-todorovic/renovate-sharedconfig:docker=1.0.3",
    "github>michael-todorovic/renovate-sharedconfig:helm",
    "github>michael-todorovic/renovate-sharedconfig=1.0.0",
    "github>michael-todorovic/renovate-sharedconfig"
```

However, re-reading https://github.com/renovatebot/renovate/blob/main/docs/development/shareable-configs.md makes me think my kind of syntax is not good enough. The tag is meant for a repo/package so I think it's better to set it close to the repo/package name.

What do you think about:
```
# git-based
github>abc/foo=tag
github>abc/foo=tag:presetName
github>abc/foo/presetPath=tag:presetName

# non-scoped
somepackagename=tag:default
somepackagename=tag:default(eslint)
renovate-config-somepackagename=tag:default
renovate-config-somepackagename=tag:default(eslint)

# scoped
@somescope=tag
@somescope/somepackagename=tag
@somescope=tag:webapp
@somescope=tag(eslint)
@somescope=tag:webapp(eslint)
@somescope/somepackagename=tag:webapp
@somescope/somepackagename=tag:webapp(eslint)
```

I'm not forcing any tag syntax, people is free to version how they want. Obviously, `tag` is optional and if it's absent, we fallback to the current behavior.

**Reply by rarkins** [reply]
We don't need to support versioned npm presets - only git. That should simplify things.

**Reply by michael-todorovic** [reply]
Here we go :) https://github.com/renovatebot/renovate/pull/11565

I'll try to write the preset manager when I have a moment if no ones takes care of it :) In the meantime, I'll go with regexManager
